### PR TITLE
Make Federation connections (links) specify a name

### LINKED ISSUE #39: Make Federation connections (links) specify a name
After https://github.com/rabbitmq/rabbitmq-server/issues/104 we can make Federation connections easier to tell from those of other clients.

### PR DESCRIPTION
**Pull request by acogoluegnes**
The connection name pattern is `Federation [Upstream name] [Policy name]`.

Fixes #39

screenshot from 2016-10-13 17 56 53

### CODE REVIEW

**Comment by acogoluegnes** [comment]
A test failed, but it looks like it was a transient failure.

File: src/rabbit_federation_link_util.erl [hunk 1]
```diff
@@ -82,15 +82,36 @@ start_conn_ch(Fun, Upstream, UParams,
...
+    #upstream_params{x_or_q = Resource}) when is_record(Resource, amqqueue)->
+    Policy = Resource#amqqueue.policy,
+    PolicyName = proplists:get_value(name, Policy),
+    connection_name(UpstreamName, PolicyName);
+
+get_connection_name(_, _) ->
+    connection_name(undefined, undefined).
+
+connection_name(Upstream, Policy) when is_binary(Upstream), is_binary(Policy) ->
+    <<<<"Federation ">>/binary, Upstream/binary, <<" ">>/binary, Policy/binary>>;
```
Comment (michaelklishin) [inline src/rabbit_federation_link_util.erl]: Please make it "Federation link (upstream: …, policy: …)".

File: src/rabbit_federation_link_util.erl [hunk 2]
```diff
@@ -82,15 +82,36 @@ start_conn_ch(Fun, Upstream, UParams,
...
+
+connection_name(_, _) ->
+    <<"Federation">>.
```
Comment (michaelklishin) [inline src/rabbit_federation_link_util.erl]: "Federation link" would be clearer.

File: src/rabbit_federation_link_util.erl [hunk 3]
```diff
@@ -82,15 +82,36 @@ start_conn_ch(Fun, Upstream, UParams,
...
+open_monitor(Params, Name) ->
+    case open(Params, Name) of
         {ok, Conn, Ch} -> erlang:monitor(process, Ch),
                           {ok, Conn, Ch};
         E              -> E
     end.
 
-open(Params) ->
-    case amqp_connection:start(Params) of
+open(Params, Source) ->
```
Comment (michaelklishin) [inline src/rabbit_federation_link_util.erl]: The parameter seems to be a name, not a source.

File: src/rabbit_federation_link_util.erl (hunk 2 above)
Comment (acogoluegnes) [inline src/rabbit_federation_link_util.erl]: Done. Looks like this now:

screenshot from 2016-10-19 09 37 17

**Pull request merged**